src/Engine/Renderer/RenderObject/RenderObject.hpp

```diff
@@ -122,6 +123,10 @@ class RA_ENGINE_API RenderObject final : public Core::Utils::IndexedObject
     std::shared_ptr<const RenderTechnique> getRenderTechnique() const;
     std::shared_ptr<RenderTechnique> getRenderTechnique();
 
+    void setMaterial( const std::shared_ptr<Material>& material );
+    std::shared_ptr<const Material> getMaterial() const;
+    std::shared_ptr<Material> getMaterial();
```

Comment:
Since the getter/setter section is not a documentation section, these are not documented at all.
And even though names are explicit, everything must be documented.